### DISCUSSION TITLE
splunk release 7 (otel 0.15.1) passing service name

### DISCUSSION THREAD

**Discussion by sparam** [signalfx/splunk-otel-java#114] [Q&A]
Hi,

I have tried few options for setting service name in release version 7 (otel 0.15.1). But the APM ui shows
unknown_service:java. Can you please share a correct system property for passing service name?

Below is one example I tried.

java -javaagent:./splunk-otel-javaagent-all-v7.jar -Dotel.trace.exporter=jaeger-thrift-splunk -Dotel.exporter.jaeger.service.name=batch -Dotel.exporter.jaeger.endpoint=http://host:14268/api/traces -jar batch-processing-0.0.1-SNAPSHOT.jar

**Comment by mateuszrzeszutek** [comment]
Hi,

Try setting it using `-Dotel.resource.attributes=service.name=batch`.
I've just reproduced your issue locally, it looks like OTel jaeger-thrift span exporter ignores manually set service name and always uses one from the resource. AFAIK the move to resource attributes (and turning off the deprecated property) was supposed to happen in the next SDK release, but it seems that a bug has creeped in a bit earlier.

We'll update release notes + README accordingly.

**Reply by mateuszrzeszutek** [reply]
Anyway, thanks for catching this so quickly @sparam!
And sorry for the trouble... 😞

**Reply by sparam** [reply]
thanks. setting to resource attribute works. any reason the name gets set to service.name and in turn to resource.attributes; I have not seen a similar way of setting properties before; seems it can get confusing.

`java -javaagent:./splunk-otel-javaagent-all-v7.jar -Dotel.trace.exporter=jaeger-thrift-splunk -Dotel.resource.attributes=service.name=batch-app -Dotel.exporter.jaeger.endpoint=http://host:14268/api/traces -jar batch-processing-0.0.1-SNAPSHOT.jar`

**Reply by iNikem** [reply]
`otel.resource.attributes` is the name of the property to be set. The value that we are setting is `service.name=batch-app`. That value in turn is itself a key-value pair, `service.name` is a key and `batch-app` is a value.
